src/ip.py

```diff
@@ -185,45 +190,69 @@ def do_route_list(af):
         dev = ra[5]
       if flags.find('W') != -1:
         continue
+      if flags.find('B') >= 0:
+        blackholes.append(ra)
+        continue
       if target == 'default':
         print('default via ' + gw + ' dev ' + dev)
       else:
-        dots=target.count('.')
-        if target.find('/') == -1:
-          addr=target
-          netmask=8+dots*8
-        else:
-          [addr,netmask] = target.split('/')
-
-        if dots == 2:
-          addr = addr + '.0'
-        elif dots == 1:
-          addr = addr + '.0.0'
-        elif dots == 0:
-          addr = addr + '.0.0.0'
-
+        cidr = cidr_from_netstat_target(target)
         if re.match("link.+",gw):
-          print(addr + '/' + str(netmask)+ ' dev ' + dev + '  scope link')
+          print(cidr + ' dev ' + dev + '  scope link')
         else:
-          print(addr + '/' + str(netmask) + ' via ' + gw + ' dev ' + dev)
+          print(cidr + ' via ' + gw + ' dev ' + dev)
+  for b in blackholes:
+    target = b[0]
+    if af == 6:
+      target = re.sub('%[^ ]+/','/',target)
+      print('blackhole ' + target)
+    else:
+      cidr = cidr_from_netstat_target(target)
+      print('blackhole ' + cidr)
   return True
 
+def cidr_from_netstat_target(target):
+  dots=target.count('.')
+  if target.find('/') == -1:
+    addr=target
+    netmask=8+dots*8
+  else:
+    [addr,netmask] = target.split('/')
+
+  if dots == 2:
+    addr = addr + '.0'
+  elif dots == 1:
+    addr = addr + '.0.0'
+  elif dots == 0:
+    addr = addr + '.0.0.0'
+  return addr + '/' + str(netmask)
+
+
 def do_route_add(argv,af):
   target=argv[0]
   via=argv[1]
-  gw=argv[2]
+  options=""
+  if "blackhole"==target:
+    target=via
+    via="via"
+    gw="127.0.0.1"
+    options="-blackhole " 
+  else:
+    gw=argv[2]
   if via not in ['via','nexthop','gw','dev']:
     do_help_route()
   inet=""
   if ":" in target or af==6:
     af=6
     inet="-inet6 "
   if "dev"==via: gw = "-interface " + gw
-  return execute_cmd(SUDO + " " + ROUTE + " add " + inet + target + " " + gw)
+  return execute_cmd(SUDO + " " + ROUTE + " add " + inet + target + " " + gw + " " + options)
 
 def do_route_del(argv,af):
   target=argv[0]
   inet=""
+  if "blackhole"==target:
+    target=" ".join([ argv[1], "127.0.0.1", "-blackhole" ])
```

Comment:
Then I cheated here and didn't set it up for more options, as it was kind of sparse to start with